### PR TITLE
Music: library improvements

### PR DESCRIPTION
**Pull request by breville**
Some improvements to the music library code, including simpler enumeration functions for use by `SoundsPanel` and `PackDialog`.

### CODE REVIEW

**Review by thomasoniii — APPROVED** [review]
lgtm 🚀

File: apps/src/music/player/MusicPlayer.ts [hunk 1]
```diff
@@ -539,7 +539,7 @@ export default class MusicPlayer {
       return;
     }
 
-    for (const instrument of library.folders.filter(
+    for (const instrument of library.instruments.filter(
```
Comment (sanchitmalhotra126) [inline apps/src/music/player/MusicPlayer.ts]: I think this will need to be both `[...library.instruments, ...library.kits]`

**Review by sanchitmalhotra126 — APPROVED** [review]
Much cleaner! Thanks!

File: apps/src/music/player/MusicPlayer.ts (hunk 1 above)
Comment (breville) [reply apps/src/music/player/MusicPlayer.ts]: Great catch.  Fixed.

**Review by thomasoniii — APPROVED** [review]

File: apps/src/music/player/MusicPlayer.ts [hunk 2]
```diff
@@ -539,7 +539,7 @@ export default class MusicPlayer {
       return;
     }
 
-    for (const instrument of library.instruments.filter(
+    for (const instrument of [...library.instruments, ...library.kits].filter(
       folder => folder.type === 'instrument' || folder.type === 'kit'
     )) {
```
Comment (thomasoniii) [inline apps/src/music/player/MusicPlayer.ts]: Keeping in mind I know nothing about the data, but why is this filter necessary?

Just looking at it, it looks like it's iterating over a list of instruments and a list of kits, but then it's filtering out any folder type that isn't an instrument or a kit.

Which implies that we could have something that's not an instrument or a kit in the list of instruments and kits.

How could that happen? Is it something that should/could be prevented elsewhere?

Comment (sanchitmalhotra126) [reply apps/src/music/player/MusicPlayer.ts]: Good catch! The `filter` is old code from when we didn't separate `instrument`s and `kit`s in the library. The filter can be removed now.

Comment (breville) [reply apps/src/music/player/MusicPlayer.ts]: Thanks for the catch.  It looks like this code was removed entirely in https://github.com/code-dot-org/code-dot-org/pull/57849?

Comment (sanchitmalhotra126) [reply apps/src/music/player/MusicPlayer.ts]: Oh true, this function has been removed now since we don't need to pre-load all instrument sounds at once.

**Review by sanchitmalhotra126 — APPROVED** [review]

**Pull request merged**